xbmc/ServiceBroker.cpp

```diff
@@ -154,8 +152,7 @@ CWinSystemBase& CServiceBroker::GetWinSystem()
 
 CRenderSystemBase& CServiceBroker::GetRenderSystem()
 {
-  CRenderSystemBase &renderSystem = dynamic_cast<CRenderSystemBase&>(g_application.m_ServiceManager->GetWinSystem());
```

Comment:
this is wrong. keep it as it was. don't yo think there was a reason why it implemented the way it is?

Reply:
?

Reply:
ServiceManager manages the lifecycle of services/objects. Knowledge about internals does not belong to this role. 
The interface to RenderSystem should not even be provided by ServiceBroker but by WinSystem directly.